### PR TITLE
thread: add uv_thread_cb typedef

### PR DESCRIPTION
**Pull request by trevnorris**
To keep with convention, add a typedef for the function called that
initializes the new thread.

### CODE REVIEW

File: include/uv.h [hunk 1]
```diff
@@ -2204,8 +2204,14 @@ UV_EXTERN void uv_key_delete(uv_key_t* key);
...
-UV_EXTERN int uv_thread_create(uv_thread_t *tid,
-    void (*entry)(void *arg), void *arg);
+/*
+ * Callback that is invoked to initialize thread execution.
+ *
+ * `arg` is the same value that was passed to uv_thread_create().
+ */
+typedef void (*uv_thread_cb)(void* arg);
+
+UV_EXTERN int uv_thread_create(uv_thread_t *tid, uv_thread_cb, void *arg);
```
Comment (indutny) [inline include/uv.h]: Mind changing the signature at the implementation site too?

Comment (indutny) [inline include/uv.h]: Also, please make it `uv_thread* tid` and `void* arg` while we are here

**Comment by indutny** [comment]
Few comments, otherwise LGTM

File: include/uv.h (hunk 1 above)
Comment (trevnorris) [inline include/uv.h]: derp. feel stupid for missing the first one.

**Comment by trevnorris** [comment]
@indutny both things fixed. thanks much.

**Comment by indutny** [comment]
Thanks, landed in 32597100b81a03cc827c164ca8250fba91599593 !

**Pull request closed without merging**